api/baseapi.cpp

```diff
@@ -1891,31 +1891,44 @@ char* TessBaseAPI::GetUNLVText() {
   return result;
 }
 
-/**
- * The recognized text is returned as a char* which is coded
- * as UTF8 and must be freed with the delete [] operator.
- * page_number is a 0-based page index that will appear in the osd file.
- */
-char* TessBaseAPI::GetOsdText(int page_number) {
+bool TessBaseAPI::DetectOrientationScript(int& orient_deg, float& orient_conf, std::string& script, float& script_conf) {
```

Comment:
You should use plain pointers instead of references.
This is what all(?) the other methods which are part of the API do.
https://google.github.io/styleguide/cppguide.html#Reference_Arguments